File: java/src/main/java/club/marzipan/javacarrentals/CreateAccount.java
package club.marzipan.javacarrentals;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

public class CreateAccount {

    @NotBlank
    @Size(min = 8, max = 16, message = "length must be 8 to 16 bytes")
    public String username;

    @NotBlank
    @Size(max = 64, message = "length must be 64 bytes or less")
    @Email(message = "malformed")
    public String email;

}

File: rust/src/create_account.rs
use serde::Deserialize;
use validator::Validate;

#[derive(Validate, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateAccount {
    #[validate(length(min = 8, max = 16, message = "length must be 8 to 16 bytes"))]
    #[serde(default)]
    pub username: String,

    #[validate(email(message = "malformed"))]
    #[validate(length(max = 64, message = "length must be 64 bytes or less"))]
    #[serde(default)]
    pub email: String,
}
